forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt

```diff
@@ -127,7 +153,49 @@ class ForageTerminalSDK(
     override suspend fun checkBalance(
         params: CheckBalanceParams
     ): ForageApiResponse<String> {
-        return ForageApiResponse.Success("TODO")
+        val logger = createLogger()
+        val forageSdkInstance = forageSdk as ForageSDK
```

Comment:
Can we always safely cast this? I assume the type is `ForageSDKInterface` because we want to mock it in testing, but if we are going to straight cast it here should we just make the type `ForageSDK`?

Reply:
ff835b192bb21e08bfed956cc3cf8b8801b70dba

I used `mockito` now instead of implementing the `ForageSDKInterface` to avoid the slightly more hazardous casting in `checkBalance`.

The original challenge was with mocking `ForageSDK` via inheritance as it is (and I think it should be) a final class